torch/csrc/jit/backends/nnapi/nnapi_backend_lib.cpp

```diff
@@ -17,23 +28,116 @@ class NnapiBackend : public PyTorchBackendInterface {
     return true;
   }
 
-  // Function stub
-  // TODO: implement compile
   c10::impl::GenericDict compile(
       c10::IValue processed,
       c10::impl::GenericDict method_compile_spec) override {
-    auto handles =
-        c10::Dict<std::string, std::vector<std::tuple<std::string, int64_t>>>();
+    auto dict = processed.toGenericDict();
+
+    // Prepare weights
+    auto weights = dict.at("weights").toTensorList();
+    for (int i = 0; i < weights.size(); i++) {
+      weights.set(i, weights.get(i).contiguous());
+    }
+    dict.insert("weights", weights);
```

Comment:
This couldn't be done in preprocess because of how values are serialized?

Reply:
Actually, it probably can be done in preprocess, since the weights are in the handles dictionary. Thanks for pointing it out. I can follow up on it with moving ser_model_.